services/QuillLMS/app/services/classlink_integration/schools_fetcher.rb

```diff
@@ -3,15 +3,18 @@
 module ClasslinkIntegration
   class SchoolsFetcher < BaseFetcher
     RESOURCE = 'schools'
-    QUERY_STRING = "?filter=type='school'"
+    QUERY_STRING = "?filter=type='school'&fields=name,identifier"
 
     def api_response_selector_override = 'orgs'
 
     def transform_api_response(school_hash)
       {
         district_id:,
-        name: school_hash['name']
+        name: school_hash['name'],
+        nces_id: self.class.coerce_nces_id(school_hash['identifier'])
       }
     end
+
+    def self.coerce_nces_id(maybe_nces_id) = maybe_nces_id.to_s =~ /\A\d{12}\z/ ? maybe_nces_id : nil
```

Comment:
Another thing worth noting here is that we've run into issues in the past with some databases left-padding `0`s in NCES IDs when appropriate, and some not such that `000000000001` being recorded as `1`.  Hard to know if that'll happen in this database, but worth being aware of if we're going to validate data like this looking for a specific number of characters.

Reply:
Thanks for the heads up. The left-padding risk should be mitigated by the fact that we're doing regex matching, not integer matching here.

If they _send_ us "1" but intend "000000000001", we'll need to take action.